guiEditor/src/diagram/workbench.tsx

```diff
@@ -464,7 +502,17 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         const onPointerDown = guiControl.onPointerDownObservable.add((evt) => {
             if (!this.isUp || evt.buttonIndex > 0) return;
             if (this._forceSelecting) {
-                this.isSelected(true, guiControl);
+                console.log(guiControl.name);
+                if (!this._doubleClick && this._isMainSelectionParent(guiControl)) {
+                    this._doubleClick = guiControl;
+                    window.setTimeout(() => {
+                        this._doubleClick = null;
+                    }, 1500);

```

Comment:
why are we waiting 1.5 seconds? is it configurable?

Reply:
This is a good point. Should this be up to the user?

Reply:
i guess the main question is why do you need the timeout at all?